text/0107-construct-library-module-lifecycle.md

```diff
@@ -0,0 +1,146 @@
+        >**Note**: Feature complete does not mean 100% AWS service feature coverage, instead, it means the module owner thinks we’ve implemented the minimum features required for a delightful developer experience for modeling the service infrastructure as code.
+     * **Successful API review** - conduct an API review/developer experience demo and invite relevant stakeholders
+        * use module's README as the API review guide
+        * address any concerns about API ergonomics
+        * review all AWS Lint exceptions to ensure they are acceptable for dev preview
+        * ensure at least 90% unit test coverage for all implemented L2 constructs
+        * if the API/DevEx is rejected, make a list of necessary remediation tasks and update the module's tracking issue 
+
+## Stage 2 - Developer Preview 
+
+The goal of developer preview is to deliver a release candidate with a stable API to conduct user acceptance testing and accumulate sufficient usage to declare a module is ready for general availability. We will only make breaking changes to developer preview modules when required to address unforeseen customer use cases or issues. Since the potential for breaking changes still remains while in Developer Preview, the package name will keep the “-experimental” suffix during this stage.
+
+* **Stage 2 activities**:
+    * **Update Roadmap** - move to the CDK public roadmap “Developer Preview” column to indicate we need customer feedback on the module’s implementation
+    * **Observe** - monitor module utilization and GitHub issues filed against the experimental module
+    * **Quality Assurance** - fix all P0 bugs as they are  discovered and review all P1 bugs, making a clearly communicated fix/defer decision for each one 
+* **Exit Criteria**:
+    * **Bake time** - modules will spend at least 3 months in developer preview to accumulate sufficient usage data to gain confidence in the API’s ergonomics and customer acceptance
+    * **Usage** - module utilization in 2,000 or more stacks is considered sufficient usage. We will evaluate the language distribution of module use to determine if we need to target specific underrepresented programming language communities for feedback.
+    * **No P0 bugs** - no new P0 bugs in the past 4 weeks
+    * **Successful API/Docs review** - conduct the final API review/developer experience demo and invite relevant stakeholders
+        * review all AWS Lint exceptions and have a high bar for keeping them at GA
+        * review and sign-off module’s README by tech writer
+        * review and sign-off module's API by module owner
+        * if the API/DevEx is rejected, make a list of necessary remediation tasks and update the module's tracking issue
+
+## Stage 3 - GA
+
+The module is generally available with a backwards compatible guarantee across minor versions. 
+
+* **Stage 3 activities**:
+    * **Update Roadmap** - move to the CDK roadmap “shipped” column
```

Comment:
Ok, i'm slowly coming to realize that the roadmap is actually our bookkeeping mechanism with regards to module maturity. Is that correct? If so, where do we expose our intentions? Basically, where do we let users know that: *"we have chosen library X to move to stage Y, doing all the necessary work to make it happen"*?

Reply:
I _hope_ that I've answered this in the comments above. :)  If not, let me know and we'll go over the roadmap doc together in person.  I think showing concrete examples will help.